src/components/encounter/encounter-form-manager.ts

```diff
@@ -101,6 +109,35 @@ export class EncounterFormManager {
       return savePatientIdentifier(patientIdentifier, patient.id);
     });
   }
+
+  static preparePatientPrograms(
+    fields: FormField[],
+    patient: fhir.Patient,
+    currentPatientPrograms: Array<PatientProgram>,
+  ): Array<PatientProgramPayload> {
+    const programFields = fields.filter((field) => field.type === 'programState' && hasSubmission(field));
+    return programFields.map((field) => {
+      const programUuid = field.questionOptions.programUuid;
+      const existingProgramEnrollment = currentPatientPrograms.find((program) => program.program.uuid === programUuid);
+      if (existingProgramEnrollment) {
+        return {
```

Comment:
When transitioning someone from one state to the next, you need to retire the previous, otherwise they will have two active states in one workflow

Reply:
We actually don't have to manually do that from the client since the backend was designed to handle that for us.  So yes, the previous state is marked as "ended" by the backend API.

Reply:
Perfect